packages/commerce-sdk-react/src/components/ShopperExperience/Component/index.tsx

```diff
@@ -24,17 +119,53 @@ const ComponentNotFound = ({typeId}: ComponentType) => (
  */
 export const Component = ({component}: ComponentProps) => {
     const pageContext = usePageContext()
-    const ComponentClass = pageContext?.components[component.typeId] || ComponentNotFound
+    const {jsxParserComponents} = pageContext
     const {data, ...rest} = component
+    const {code} = data || {}
+    const isEinsteinAssistedComponent = !!code
```

Comment:
As mentioned before this should look at the `component.typeId` to determine if this is in fact the EGPT component. Otherwise this can/will get triggered by unrelated components that have this (commonly-named) property.